In [1]:
import pandas as pd
from sklearn import preprocessing

In [2]:
s=pd.read_csv('train.csv')
s.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [3]:
from sklearn import neighbors

In [4]:
l_encoder=preprocessing.LabelEncoder()
encoded_g=l_encoder.fit_transform(s['Sex'])
s['Sex']=encoded_g
s['Embarked']=l_encoder.fit_transform(s['Embarked'])

In [5]:
s=s.drop(['Cabin','PassengerId','Name','Ticket'],axis=1)
s.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2

# Implementing KNN

In [7]:
def knn(a,b):
    from sklearn.model_selection import train_test_split
    from sklearn.metrics import confusion_matrix
    X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)
    sc1=0
    for k in range(1,237):
        knn=neighbors.KNeighborsClassifier(n_neighbors=k)
        sc=knn.fit(X_train,y_train).score(X_test,y_test)
        y_pred=knn.predict(X_test)
        confusion_matrix(y_test,y_pred)
        m=max(sc,sc1)
        sc1=m
    return sc1

# Pclass vs  remaining attributes

In [8]:
y=s['Pclass']
X=s.drop(['Pclass'],axis=1)

In [9]:
p1=knn(X,y)
p1

0.8913857677902621

### Pclass vs remaining attributes 89.1%

# Sibsp vs remaining attributes

In [11]:
y=s['SibSp']
X=s.drop(['SibSp'],axis=1)
p1=knn(X,y)
p1

0.7303370786516854

### SibSp vs remaining attributes 73%

# Parch vs remaining attributes

In [12]:
y=s['Parch']
X=s.drop(['Parch'],axis=1)
p1=knn(X,y)
p1

0.797752808988764

### Parch vs remaining attributes 79.7%

# Embarked vs remaining attributes

In [14]:
y=s['Embarked']
X=s.drop(['Embarked'],axis=1)
p1=knn(X,y)
p1

0.7790262172284644

### Embarked vs remaining attributes 77.9%

# Gender vs remaining attributes

In [15]:
y=s['Sex']
X=s.drop(['Sex'],axis=1)
p1=knn(X,y)
p1

0.6928838951310862

### Gender vs remaining attributes 69.2%

# Survived vs remaining attributes

In [16]:
y=s['Survived']
X=s.drop(['Survived'],axis=1)
p1=knn(X,y)
p1

0.704119850187266

### Survived vs remaining attributes 70.4%